controllers/csiaddons/csiaddonsnode_controller.go

```diff
@@ -184,6 +196,59 @@ func (r *CSIAddonsNodeReconciler) removeFinalizer(
 	return nil
 }
 
+// resolveEndpoint parses the endpoint and returned a string that can be used
+// by GRPC to connect to the sidecar.
+func (r *CSIAddonsNodeReconciler) resolveEndpoint(ctx context.Context, rawURL string) (string, error) {
+	namespace, podname, port, err := parseEndpoint(rawURL)
+	if err != nil && errors.Is(err, errLegacyEndpoint) {
+		return rawURL, nil
+	} else if err != nil {
+		return "", err
+	}
+
+	pod := &corev1.Pod{}
+	err = r.Client.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
```

Comment:
parseEndpoint can return empty namespace, if the namespace is `""` empty the `r.Client.Get` will fail. we need to set the namespace to `default` if not set

Reply:
`default` is just a namespace like any other, I do not think it should be used if none it set. Not sure what is better, fail the `Get()` with an error, or report the error before calling it?

Reply:
done, added more checks